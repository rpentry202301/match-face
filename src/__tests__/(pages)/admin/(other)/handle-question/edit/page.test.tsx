import { render, screen, fireEvent } from "@testing-library/react";
import EditQuestionPage from "@/app/(pages)/admin/(other)/handle-question/edit/[id]/page";
import React from "react";
import "@testing-library/jest-dom";
import userEvent from "@testing-library/user-event";

describe("質問・回答例編集画面テスト", () => {
  beforeEach(() => {
    jest.clearAllMocks();
    render(<EditQuestionPage params={{id:'1'}}/>);
  });
  afterAll(() => {
    jest.clearAllMocks();
  });
  
  it("スナップショット", async () => {
    const { container } = render(<EditQuestionPage params={{id:'1'}}/>);
    expect(container).toMatchSnapshot();
  });

  it("記述式質問追加ボタンクリック", async () => {
    const addWriteButton = screen.getByTestId("addWriteButton");
    fireEvent.click(addWriteButton);
    expect(screen.getByTestId("write_4")).toHaveTextContent("Q4");
  });

  it("選択式質問追加ボタンクリック", async () => {
    const addSelectButton = screen.getByTestId("addSelectButton");
    fireEvent.click(addSelectButton);
    expect(screen.getByTestId("select_4")).toHaveTextContent("Q4");
  });

  it("一覧へ戻るボタンリンク", async () => {
    const backListButton = screen.getByTestId("backList");
    expect(backListButton).toHaveAttribute("href", "/admin/handle-question");
  });

  it("バリデーションテスト（記述式）", async () => {
    const addWriteButton = screen.getByTestId("addWriteButton");
    const sendButton = screen.getByTestId("sendButton");
    const errMsg = screen.getByTestId("errMsg");
    fireEvent.click(addWriteButton);
    fireEvent.click(sendButton);
    expect(errMsg).toHaveTextContent("未入力の項目があります");
  });

  it("バリデーションテスト（選択式）", async () => {
    const addSelectButton = screen.getByTestId("addSelectButton");
    const sendButton = screen.getByTestId("sendButton");
    const errMsg = screen.getByTestId("errMsg");
    fireEvent.click(addSelectButton);
    fireEvent.click(sendButton);
    expect(errMsg).toHaveTextContent("未入力の項目があります");
  });
});
