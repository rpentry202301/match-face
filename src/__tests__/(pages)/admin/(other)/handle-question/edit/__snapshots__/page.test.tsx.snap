// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`質問・回答例編集画面テスト スナップショット 1`] = `
<div>
  <div
    class="flex flex-col items-center my-20 w-4/5 mx-auto h-auto"
  >
    <div
      class="w-2/3"
    >
      <div
        class="flex items-end justify-between border-b-2 border-black pb-3"
      >
        <h1
          class="text-3xl"
        >
          質問・回答例の編集
        </h1>
        <p>
          登録日:
          /NaN/NaN
        </p>
      </div>
      <div
        class="my-8"
      >
        <h2
          class="text-2xl"
        >
          案件名：
        </h2>
        <p
          class="border p-3 mt-2"
        />
      </div>
    </div>
    <div
      class="flex flex-col"
    >
      <div
        class="flex justify-around m-3"
      >
        <button
          class="bg-white hover:bg-gray-100 active:bg-gray-200 active:translate-y-0.5 active:shadow-sm py-2 px-4 rounded-full border shadow-md m-10"
          data-testid="addWriteButton"
        >
          記述質問追加
        </button>
        <button
          class="bg-white hover:bg-gray-100 active:bg-gray-200 active:translate-y-0.5 active:shadow-sm py-2 px-4 rounded-full border shadow-md m-10"
          data-testid="addSelectButton"
        >
          選択質問追加
        </button>
      </div>
      <p
        class="text-center text-red"
        data-testid="errMsg"
      />
      <div
        class="flex m-3"
      >
        <a
          data-testid="backList"
          href="/admin/handle-question"
        >
          <div>
            <button
              class="bg-orange drop-shadow-md hover:saturate-150 active:drop-shadow-none active:shadow-inner hover:bg-depp-orange active:bg-depp-orange active:translate-y-0.5 text-white p-2 w-40 h-auto text-lg m-10 rounded-none py-5 flex items-center justify-center"
            >
              &lt; 一覧へ戻る
            </button>
          </div>
        </a>
        <div>
          <button
            class="bg-orange drop-shadow-md hover:saturate-150 active:drop-shadow-none active:shadow-inner hover:bg-depp-orange active:bg-depp-orange active:translate-y-0.5 text-white p-2 w-40 h-auto text-lg m-10 rounded-none py-5 flex items-center justify-center"
            data-testid="sendButton"
          >
            保存する
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
